app/models/general_info.rb

```diff
@@ -44,9 +43,27 @@ class GeneralInfo
             numericality: { only_integer: true, greater_than: 0, less_than_or_equal_to: 20,
                             message: "should be a positive number (1-20)" }
   validates :ori, ori: true
+  validates :address, presence: true
+  validate :incident_unique?
 
   delegate :incident_id, to: :incident
 
+  def incident_unique?
+    return false if ori.blank? || city.blank? || address.blank? || incident_time_str.blank? || incident_date_str.blank?
+    general_infos = GeneralInfo.where(ori: ori, city: city, address: address,
+                                      incident_time_str: incident_time_str,
+                                      incident_date_str: incident_date_str)
+    gi = general_infos.first
```

Comment:
This still has the issue that I mentioned before – what if the first `GeneralInfo` in `general_infos` is the current one? (This could happen if the current GeneralInfo page has already been saved and is now being edited).

We need to check all `general_infos`, not just the first.

So something like this ought to work in place of lines 56-64
```ruby
if general_infos.any? {|gi| gi.id != id && !gi.incident.try(:deleted)}
  errors.add(:address, "there is already another incident with this same date, time, address, city, and ORI - are you sure you didn't fill out this incident report already?")
end
```